graphs/graph_modules/data_fetcher.py

```
@@ -0,0 +1,270 @@
+                "apikey": self.config["TAUTULLI_API_KEY"],
+                "cmd": cmd,
+                "start_date": start_date.strftime("%Y-%m-%d"),
+                "end_date": now.strftime("%Y-%m-%d"),
+                **params
+            }
+
+            # Remove None values from params
+            api_params = {k: v for k, v in api_params.items() if v is not None}
+
+            async with aiohttp.ClientSession() as session:
+                async with session.get(self.config["TAUTULLI_URL"], params=api_params) as response:
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)